### PR TITLE
creating login page

### PR DESCRIPTION
**Pull request by HiQoSolutionsAI**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,13 +11,17 @@
...
     "@angular/common": "^14.1.0",
     "@angular/compiler": "^14.1.0",
     "@angular/core": "^14.1.0",
+    "@angular/fire": "^7.4.1",
     "@angular/forms": "^14.1.0",
+    "@angular/material": "^14.2.0",
     "@angular/platform-browser": "^14.1.0",
     "@angular/platform-browser-dynamic": "^14.1.0",
     "@angular/router": "^14.1.0",
+    "firebase": "^9.9.3",
```
Comment (EkaterinaSaverina) [inline package.json]: Why we need this package?

File: src/_mixins.scss [hunk 2]
```diff
@@ -0,0 +1,41 @@
+@mixin fonts-style($font-size: 16px, $font-weight: 500, $line-height: 26px, $color: $basic-state-text) {
+  color: $color;
+  font-weight: $font-weight;
+  font-size: $font-size;
+  line-height: $line-height;
+}
+
```
Comment (EkaterinaSaverina) [inline src/_mixins.scss]: Please add letter-spacing property as well

File: src/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+$primary:#3f51b5;
+$warn:#ff0000;
+$basic-state-background:#eceff1;
+$basic-state-text:#fff;
+$accent-state-text:#000;
```
Comment (EkaterinaSaverina) [inline src/_variables.scss]: wrong formating

File: src/app/auth/login/login-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { LoginComponent } from './login.component';
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login-routing.module.ts]: Please add an empty line bitween core and internal imports

File: src/app/auth/login/login.component.scss [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  width: 100%;
+
+  &-wrapper {
+    display: flex;
+    flex-direction: column;
+    width: 540px;
+    padding: 35px;
+    margin-left: 15px;
+    margin-right: 15px;
+    background-color: #eceff1;
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login.component.scss]: Use a variable for colors

File: src/app/auth/login/login.component.scss [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+  &-up {
+    margin-left: -4px !important;
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+}
+
+.title {
+  font-size: 36px;
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login.component.scss]: Use a mixin for font styles

File: src/app/auth/login/login.component.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    });
+  }
+
+  registration(): void {
+    const user: IUser = {
+      id: '',
+      username: this.loginForm.get('username')!.value,
+      email: this.loginForm.get('email')!.value,
+      profilePhoto: '',
+      createdOn: 0,
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login.component.ts]: Use Timestamp here

File: src/models/board.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import { IModel } from './model';
+
+export interface IBoard extends IModel {
+  Title: string;
```
Comment (EkaterinaSaverina) [inline src/models/board.ts]: A property name should begin from a small letter

File: src/app/services/user/user.service.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(
+    private auth: AngularFireAuth,
+    private db: AngularFirestore
+  ) {
+    // this.user = this.auth.user.pipe(
+    //   switchMap((user) =>(!user) ? null : this.db.collection('Users').doc<IUser>(user.uid).valueChanges()
+    //   )
+    // );
+  }
+}
```
Comment (EkaterinaSaverina) [inline src/app/services/user/user.service.ts]: Please do not push dev code

File: src/app/services/auth/auth.service.ts [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(
+    private auth: AngularFireAuth,
+    private db: AngularFirestore,
+    private router: Router
+  ) {}
+
+  signIn(email: string, password: string) {
```
Comment (EkaterinaSaverina) [inline src/app/services/auth/auth.service.ts]: Please add a return type

File: src/app/services/auth/auth.service.ts [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    return this.auth.signInWithEmailAndPassword(email, password).then(() => {
+      this.router.navigate(['/board']);
+    });
+  }
```
Comment (EkaterinaSaverina) [inline src/app/services/auth/auth.service.ts]: Please use async await syntax

File: src/app/auth/login/login.component.scss [hunk 12]
```diff
@@ -1,3 +1,6 @@
+@import "../../../mixins";
+@import "../../../variables";
+
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login.component.scss]: Please use imports for scss the following way:
@import "mixins";

File: src/app/auth/login/login.component.scss [hunk 13]
```diff
@@ -51,13 +56,15 @@
 }
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login.component.scss:56]: Please use a variable for colors

File: src/app/auth/login/login.component.scss [hunk 14]
```diff
@@ -51,13 +56,15 @@
...
 
 .sub-title {
-  margin-top: 10px;
-  font-size: 24px;
+  @include fonts-style(24px, 600, 12px, 1px, $accent-state-text);
+
```
Comment (EkaterinaSaverina) [inline src/app/auth/login/login.component.scss]: Please separate font styles from component ones in the common _typography.scss file

File: src/app/core/services/auth.service.ts [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+      user.email,
+      password
+    );
+
+    user.createdOn = new Date();
+    if (userCredential.user?.uid) {
+      user.id = userCredential.user.uid;
+    }
+
+    await this.db.collection('users').doc(userCredential.user?.uid).set(user);
```
Comment (EkaterinaSaverina) [inline src/app/core/services/auth.service.ts]: Please separate database logic from a component one

File: src/app/core/services/auth.service.ts [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    this.router.navigate(['/board']);
```
Comment (EkaterinaSaverina) [inline src/app/core/services/auth.service.ts]: Move this code to component

**Review by EkaterinaSaverina — CHANGES_REQUESTED** [review]

File: src/app/auth/login/login.component.scss (hunk 14 above)
Comment (HiQoSolutionsAI) [reply src/app/auth/login/login.component.scss]: I'll do it in a separate pull request

**Review by EkaterinaSaverina — APPROVED** [review]

**Pull request merged**